DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -0,0 +1,457 @@
+  //--- Trigger 
+  //-------------------------------
+  edm::Handle<edm::TriggerResults> hTrgRes;
+  iEvent.getByToken(fHLTTag_token,hTrgRes);
+  if(!hTrgRes.isValid()) return;
+
+  edm::Handle<trigger::TriggerEvent> hTrgEvt;
+  iEvent.getByToken(fHLTObjTag_token,hTrgEvt);
+
+  const edm::TriggerNames &triggerNames = iEvent.triggerNames(*hTrgRes);
+  bool config_changed = false;
+  if(fTriggerNamesID != triggerNames.parameterSetID()) {
+    fTriggerNamesID = triggerNames.parameterSetID();
+    config_changed  = true;
+  }
+  if(config_changed) {
+    initHLT(*hTrgRes, triggerNames);
+  }
+
+  TriggerBits triggerBits;
+  for(unsigned int irec=0; irec<fTrigger->fRecords.size(); irec++) {
+    if(fTrigger->fRecords[irec].hltPathIndex == (unsigned int)-1) continue;
+    if(hTrgRes->accept(fTrigger->fRecords[irec].hltPathIndex)) {
+      triggerBits [fTrigger->fRecords[irec].baconTrigBit] = 1;
+    }
+  }
+  //if(fSkipOnHLTFail && triggerBits == 0) return;
+
+  // Trigger requirement
+  if(!isMuonTrigger(*fTrigger, triggerBits)) return;
+
+  //-------------------------------
+  //--- Muons and Tracks 
+  //-------------------------------
+  edm::Handle<reco::MuonCollection> hMuonProduct;
+  iEvent.getByToken(fMuonName_token,hMuonProduct);
+  if(!hMuonProduct.isValid()) return;
+
+  edm::Handle<reco::TrackCollection> hTrackProduct;
+  iEvent.getByToken(fTrackName_token,hTrackProduct);
+  if(!hTrackProduct.isValid()) return;
+
+  TLorentzVector vTag(0.,0.,0.,0.);
+  TLorentzVector vProbe(0.,0.,0.,0.);
+  TLorentzVector vTrack(0.,0.,0.,0.);
+
+  // Tag loop
+  for(auto const & itMu1 : *hMuonProduct) {
+
+    float pt1  = itMu1.muonBestTrack()->pt();
+    float eta1 = itMu1.muonBestTrack()->eta();
+    float phi1 = itMu1.muonBestTrack()->phi();
+    float q1   = itMu1.muonBestTrack()->charge();
+
+    // Tag selection: kinematic cuts, lepton selection and trigger matching
+    if(pt1        < PtCutL1_)  continue;
+    if(fabs(eta1) > EtaCutL1_) continue;
+    if(!(passMuonID(itMu1, *pv, IDType_) && passMuonIso(itMu1, IsoType_, IsoCut_))) continue;
+    if(!isMuonTriggerObj(*fTrigger, TriggerTools::matchHLT(eta1, phi1, fTrigger->fRecords, *hTrgEvt))) continue;
+
+    vTag.SetPtEtaPhiM(pt1, eta1, phi1, MUON_MASS);
+
+    // Probe loop over muons
+    for(auto const & itMu2 : *hMuonProduct) {
+      if(&itMu2 == &itMu1) continue;
+
+      float pt2  = itMu2.muonBestTrack()->pt();
+      float eta2 = itMu2.muonBestTrack()->eta();
+      float phi2 = itMu2.muonBestTrack()->phi();
+      float q2   = itMu2.muonBestTrack()->charge();
+
+      // Probe selection: kinematic cuts and opposite charge requirement
+      if(pt2        < PtCutL2_)  continue;
+      if(fabs(eta2) > EtaCutL2_) continue;
+      if(q1 == q2)               continue;
+
+      vProbe.SetPtEtaPhiM(pt2, eta2, phi2, MUON_MASS);
+
+      // Mass window
+      TLorentzVector vDilep = vTag + vProbe;
+      if((vDilep.M() < MassMin_) || (vDilep.M() > MassMax_)) continue;
```

Comment:
please make a variable for vDilep.M() to avoid redundant calculations- you use it many times.